contracts/event/LemonadeEventAttestation.sol

```diff
@@ -0,0 +1,179 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.19;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@ethereum-attestation-service/eas-contracts/contracts/IEAS.sol";
+import "@ethereum-attestation-service/eas-contracts/contracts/resolver/SchemaResolver.sol";
+import "@ethereum-attestation-service/eas-contracts/contracts/resolver/examples/AttesterResolver.sol";
+
+import "./ILemonadeEventAttestation.sol";
+import "./resolvers/EventHostSchemaResolver.sol";
+import "./resolvers/TicketIssuingSchemaResolver.sol";
+
+contract Event {}
+
+contract LemonadeEventAttestation is
+    OwnableUpgradeable,
+    ILemonadeEventAttestation
+{
+    using ECDSA for bytes;
+    using ECDSA for bytes32;
+
+    bytes32 public eventCreatorSchemaId;
+    bytes32 public eventDetailSchemaId;
+    bytes32 public eventCohostSchemaId;
+    bytes32 public ticketTypeSchemaId;
+    bytes32 public ticketSchemaId;
+
+    IEAS internal eas;
+
+    event EventCreated(
+        address eventAddress,
+        address creator,
+        bytes32 attestation
+    );
+
+    function initialize(address _eas) public initializer {
+        __Ownable_init();
+
+        eas = IEAS(_eas);
+
+        _initSchemas();
+    }
+
+    function registerEvent() external payable {
+        address creator = _msgSender();
+
+        Event event_ = new Event();
+
+        address eventAddress = address(event_);
+
+        AttestationRequestData memory data = AttestationRequestData(
```

Comment:
Would like to see what these parameters are, could you link me where this function is defined?

Reply:
```
/// @notice A struct representing the arguments of the attestation request.
struct AttestationRequestData {
    address recipient; // The recipient of the attestation.
    uint64 expirationTime; // The time when the attestation expires (Unix timestamp).
    bool revocable; // Whether the attestation is revocable.
    bytes32 refUID; // The UID of the related attestation.
    bytes data; // Custom attestation data.
    uint256 value; // An explicit ETH amount to send to the resolver. This is important to prevent accidental user errors.
}
```
@ChrisLahaye 
you can find this in `node_modules/@ethereum-attestation-service/eas-contracts/contracts/IEAS.sol`